Earthfile

```
@@ -1,53 +1,70 @@
-VERSION 0.7
-ARG gradle_version=8.2.1
-FROM gradle:${gradle_version}-jdk17
-RUN apt-get update && apt-get install -y \
-  zip \
-  && rm -rf /var/lib/apt/lists/*
-ARG --global version=0.0.0
+VERSION --pass-args --global-cache --use-function-keyword 0.7
+PROJECT earthly-technologies/core
+IMPORT github.com/earthly/lib/gradle:40a4041e3044e99cf192d05d4d1620a380d5ddc8 AS gradle
+ARG --global gradle_version=8.2.1
 ARG --global bundle="github.com/earthly/earthfile-grammar+export/"
-COPY settings.gradle.kts build.gradle.kts ./
-COPY scripts scripts
-COPY src src
+FROM gradle:${gradle_version}-jdk17
 
 GET_BUNDLE:
-  COMMAND
+  FUNCTION
   COPY $bundle build/
   RUN scripts/bundle.sh build/earthfile-syntax-highlighting
 
-dist:
+install:
+  RUN apt-get update && apt-get install -y \
+    zip \
+    && rm -rf /var/lib/apt/lists/*
+  COPY settings.gradle.kts build.gradle.kts ./
+  # Sets $EARTHLY_GRADLE_USER_HOME_CACHE and $EARTHLY_GRADLE_PROJECT_CACHE
+  DO gradle+GRADLE_GET_MOUNT_CACHE
+
+src:
+  ARG --required version
+  FROM +install
+  COPY src src
+  COPY scripts scripts
   DO +GET_BUNDLE
   RUN sed -i 's^0.0.0^'"$version"'^g' ./build.gradle.kts
```

So that a new version won't break the cache
```suggestion
src:
  FROM +install
  COPY src src
  COPY scripts scripts
  DO +GET_BUNDLE
  ARG --required version
  RUN sed -i 's^0.0.0^'"$version"'^g' ./build.gradle.kts
```